packages/common/stakeApy.ts

```diff
@@ -1,22 +1,48 @@
 import { SOL_API_HOST } from './constants';
+import { ErrorWrapper } from '../core/src/utils/errorWrapper';
+import { ERROR_CODE } from '../core/src/constants/errors';
+
+export const STATIC_DEFAULT_APY = 9.35; // TODO think
+
+const { NO_APY_DATA } = ERROR_CODE;
+
+type ApyData = {
+  apy: number;
+  apr: number;
+  intervalPrices: {
+    beginEpoch: number;
+    endEpoch: number;
+  };
+};
 
 type StakeApyResponse = {
   data: {
-    apy: number;
+    lastEpoch: ApyData;
+    twoWeeks: ApyData;
+    oneMonth: ApyData;
+    threeMonth: ApyData;
+    sinceLaunch: ApyData;
   };
 };
 
-export const STATIC_DEFAULT_APY = '5.74'; // TODO think
+export const getMaxApy = (data: StakeApyResponse['data']) => {
+  return Object.values(data).reduce((max, curr) => {
+    if (max.apy < curr.apy) {
+      return curr;
+    }
 
-export const getStakeApy = async () => {
-  try {
-    const resp = await fetch(`${SOL_API_HOST}/v1/apy?since_launch`, { mode: 'cors' });
-    const {
-      data: { apy },
-    } = (await resp.json()) as StakeApyResponse;
+    return max;
+  });
+};
 
-    return apy ? apy.toFixed(2) : STATIC_DEFAULT_APY;
-  } catch {
-    return STATIC_DEFAULT_APY;
-  }
+export const getStakeApy = async (): Promise<StakeApyResponse['data'] & { max: ApyData }> => {
+  return fetch(`${SOL_API_HOST}/v1/apy/complete`, { mode: 'cors' })
+    .then<StakeApyResponse>((res) => res.json())
+    .then(({ data }) => ({
+      max: getMaxApy(data),
+      ...data,
+    }))
+    .catch(() => {
+      throw new ErrorWrapper({ code: NO_APY_DATA });
```

Comment:
Why don't you want add error field here?

```
.catch((error) => {
      throw new ErrorWrapper({ error, code: NO_APY_DATA });
}
```

Reply:
oh yeah, thanks

Reply:
done